### PR TITLE
feat[contracts]: Add support for EIP-3074

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
WIP. Adds support for EIP-3074. You can find the full spec here: https://eips.ethereum.org/EIPS/eip-3074.

Remaining tasks:
- [ ] Add lots of comments
- [ ] Add unit tests for ovmAUTH
- [ ] Add unit tests for ovmAUTHCALL
- [ ] Add integration tests for ovmAUTH => ovmAUTHCALL

### CODE REVIEW

File: packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol [hunk 1]
```diff
@@ -799,6 +805,80 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
...
+        bytes32 _r,
+        bytes32 _s
+    )
+        override
+        public
+        returns (
+            address _authorized
+        )
+    {
+        // QUESTION: Should we throw here if _yParity != 0 or 1?
```
Comment (lightclient) [inline packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:825]: Technically, the EIP specifies:
> if the signature is instead invalid (e.g. yParity != 0 or 1), authorized is reset to an unset value.

Comment (lightclient) [reply packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:825]: doh, now seeing the next question

Comment (smartcontracts) [reply packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:825]: kk will just reset and exit early in this case!

Comment (lightclient) [reply packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:825]: `ecrecover` should fail if `yParity` is invalid, so i don't think you need to do anythin

File: packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol [hunk 2]
```diff
@@ -799,6 +805,80 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
...
+        );
+
+        address signer = ecrecover(
+            messageHash,
+            _yParity + 27,
+            _r,
+            _s
+        );
+
+        // QUESTION: EIP-3074 specifies that if the signature is "invalid," `authorized` is reset.
```
Comment (lightclient) [inline packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol:844]: Yep!

**Comment by snario** [comment]
Closing for now.

**Pull request closed without merging**